GitUI/Avatars/FileSystemAvatarCache.cs

```diff
@@ -29,7 +29,14 @@ public FileSystemAvatarCache(IAvatarProvider inner, IFileSystem? fileSystem = nu
             string cacheDir = AppSettings.AvatarImageCachePath;
             string path = Path.Combine(cacheDir, $"{email}.{imageSize}px.png");
 
-            Image image = ReadImage() ?? await _inner.GetAvatarAsync(email, name, imageSize);
+            Image image = ReadImage();
+
+            if (image is not null)
+            {
+                return image;
+            }
+
+            image = await _inner.GetAvatarAsync(email, name, imageSize);
```

Comment:
Do we need a cancellation token here?

Reply:
I don't know all the places where this code is called from, but from the one where it is call in this PR (so the avatar column in the grid), no because the call is not awaited (`.FileAndForget()`) and the screen is updated when the task finish blocking nothing...